libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1124 @@
+        cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+        sub_cmd : static_cast<uint8_t>(GimbalSubCmd::GimbalSubCmd_PlaneControlPara),
+        data : (uint8_t*)&mode
+    };
+
+    // send targets to gimbal
+    return send_packet(frame_to_send);
+}
+
+// send reference of the joystick rocker for moving the gimbal or tracking window
+// when gimbal runs in tracking mode this command will move the tracking window for selecting the target
+// when in other mode(except angle mode) this command will set the rates of the yaw axis and pitch axis(±500 map to max/min rate speed)
+bool AP_Mount_Tusuav::send_stick_pitch_yaw(int16_t pitch,int16_t yaw)
+{
+    // gimbal does not support lock in angle control mode
+    if (!set_lock(false)) {
+        return false;
+    }
+
+    if (pitch > 500) {
+		pitch = 500;
+	}
+	else if (pitch < -500) {
+		pitch = -500;
+	}
+    if (yaw > 500) {
+    	yaw = 500;
+	}
+	else if (yaw < -500) {
+		yaw = -500;
+	}
+
+    const int16_t value[2] = {pitch,yaw};
+    // fill in packet
+    const FrameType frame_to_send{
+        .flag={0},
+        data_len : sizeof(value),
+        cmd : static_cast<uint8_t>(GimbalCmd::MasCmd_Gimbal),
+        sub_cmd : static_cast<uint8_t>(GimbalSubCmd::gimbal_single_cmd_stick_control),
+        data : (uint8_t*)&value[0]
+    };
+    debug("sticking %d %d", value[0], value[1]);
+    // send targets to gimbal
+    return send_packet(frame_to_send);
+}
+
+// send vehicle attitude and position to gimbal
+bool AP_Mount_Tusuav::send_m_ahrs()
+{
+    // get current location
+    Location loc;
+    int32_t alt_amsl_cm = 0;
+    if (!AP::ahrs().get_location(loc) || !loc.get_alt_cm(Location::AltFrame::ABSOLUTE, alt_amsl_cm)) {
+        debug("tus ahrs fault:%d %d %d",(int)loc.lat,(int)loc.lng,(int)loc.alt); 
+
+    }
+
+    // get vehicle velocity in m/s in earth frame
+    Vector3f vel_NED;
+    IGNORE_RETURN(AP::ahrs().get_velocity_NED(vel_NED));
+    Vector3f acc_NED;
+    acc_NED = AP::ahrs().get_accel_ef();
+    Vector3f gyro_NED;
+    gyro_NED = AP::ahrs().get_gyro();
+
+// _gps_update_count will increase and send to gimbal if gps fixed 2D or better
+    #if AP_GPS_ENABLED
+    if(AP::gps().status() >= AP_GPS::GPS_OK_FIX_2D){
+        uint32_t gps_last_time = AP::gps().last_message_time_ms();
+        if (_gps_last_update_ms != gps_last_time) {
+            _gps_last_update_ms = gps_last_time;
+            if (++_gps_update_count >= 255) {
+                _gps_update_count = 0;
+            }
+        }
+    }
+    else{
+        _gps_update_count = 0;
+    }
+    #else
+    _gps_update_count = 0;
+    #endif
+
+ // get vehicle yaw in the range 0 to 360
+    MessageAhrs::FcStatus fc_status;
+
+    if (AP_Param::check_frame_type(AP_PARAM_FRAME_COPTER)) {
+        fc_status.bits.aircraft_type = static_cast<uint8_t>(AircraftType::Aircraft_Copter);
+    } else {
+        fc_status.bits.aircraft_type = static_cast<uint8_t>(AircraftType::Aircraft_Vtol);
```

Comment:
This doesn't look right.

Not knowing what this field is used for, it's hard to say what you should be doing here, but `Aircraft_Vtol` isn't correct for eg. a Rover.